File: Python3/프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    N = len(elements)
    sum = 0
    answer = set()
    for i in range(N):
        sum = 0
        for j in range(N):
            sum += elements[(i + j) % N]
            answer.add(sum)
        
    return len(answer)

File: C++/프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <iostream>
#include <vector>
#include <set>

using namespace std;


int solution(vector<int> elements) {
    int N = elements.size();
    
    int sum;
    set<int> nset;
    vector<int> numList;
    for(int i = 0; i < N; i++) {
        sum = 0;
        for(int j = 1; j <= N; j++) {
            sum += elements[(i + j) % N];
            nset.insert(sum);
        }
    }
    
    return nset.size();
}